rundeckapp/grails-app/domain/rundeck/Option.groovy

```diff
@@ -367,6 +387,27 @@ public class Option implements Comparable{
         return opt
     }
 
+    Set<String> getValues(){
+        if(valuesList){
+            convertValuesList()
+        }
+        if(values && sortValues){
+            sortValuesList()
+        }
+        values
+    }
+
+    void sortValuesList(){
+        def numericValues = values.findAll { it.isNumber() }.collect {it.toDouble()}
+        if(numericValues.size()==values.size()){
+            def stringNumericList= values.findAll { it.isNumber() }.collect {new StringNumericSort(it, it.toDouble())}
+            stringNumericList.sort{ a,b -> a.numericValue <=> b.numericValue }
+            values = stringNumericList.collect{it.strValue}
+        }else{
+            values = values.sort()
```

Comment:
if `Set<String> values` is not a transient, then modifying it is going to persist the value to the DB.  

I think you should not write to the `values` field anymore.  Instead just return the converted values as a List.

When an Option/Job is created/updated, the `values` field should be set to `null` instead